Sources/Sentry/SentryClient.m

```diff
@@ -412,7 +406,9 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         return nil;
     }
 
-    if (NO == [self checkSampleRate:self.options.sampleRate]) {
+    // Transactions have their own sampleRate
+    if (![event isKindOfClass:[SentryTransaction class]] &&
```

Comment:
m: event has a property called `type` that we check if is "transaction" in other parts of the project. Shouldn't this be the same here?

Reply:
Good catch thanks 🙏